### ISSUE TITLE
applyDistortion does not propagate precision. 

### ISSUE THREAD

**Issue by Kelvinrr** [USGS-Astrogeology/usgscsm#244]
`applyDistortion` in Distortion.cpp currently uses the default value for desirePrecision. ISIS3 uses pixel pitch which CSM does not have access too. We should probably propagate the desired ground precision and convert that to a precision in focal plane space.

**Comment by scsides** [comment]
Ground precision can very drastically within an image. Consider a flyby image where the whole disk is visible, or a wide angle camera with a FOV near 180deg. The distortion calculations only involve the focal plane why bring the ground into this? If we are going to propagate something why not the pixel pitch or use the coefficients for TransX,Y. These are usually the pixel pitch or very near to it.

**Comment by jessemapel** [comment]
A couple thoughts:

1. Using a fixed fraction of the pixel pitch will probably work in most cases, but does not necessarily fit well with the CSM approach of requesting things within a user-provided precision. If we want to do this, the easiest way would be to compute the diagonal pixel size from the pixel2focal affine transformations. Taking the norm of the linear portion of the transformation (The determinant of the 2x2 matrix consisting of the second and third coefficients) will give us the diagonal detector element size.

**Comment by oleg-alexandrov** [comment]
I've noticed an issue with CTX data which is likely related to this. In the ground-to-image calculation, first an iterative scheme takes place which finds the correct line. No distortion is assumed. That scheme runs until the error goes below desired precision.

Then distortion is applied to the computed line and sample, another error is evaluated, and now the error is way above what was asked by the caller. 

If this last error is the correct one, maybe the distortion operation should happen during the iterative scheme. I will attach below this a testcase (this has the model state, not the ISD).

**Comment by jessemapel** [comment]
Adding iterative distortion inversion is a non-trivial amount of overhead. We could instead "portion" the precision so that we iterate until half the precision when doing the line search and then iterate until half the precision in the distortion inversion.

**Comment by oleg-alexandrov** [comment]
The testcase for this is now in the usgscsm repo, and is named tests/data/ctxLineScan.json.

**Comment by oleg-alexandrov** [comment]
@acpaquette : Is this related to https://github.com/USGS-Astrogeology/usgscsm/pull/436?

**Comment by acpaquette** [comment]
@oleg-alexandrov That last comment from Jesse does seem related but the overhead seems negligible? Granted 5 seconds isn't negligible but does fix the problem we were seeing in KaguyaTC. I think coming back to this as a potential fix to speed up the fix applied in #436 would be a good idea

**Comment by oleg-alexandrov** [comment]
Ah, right. This original issue as filed was about propagation of precision, not modeling the distortion in ground-to-image, which  is what you fixed.

**Comment by acpaquette** [comment]
Right, and I think the case may be that the original change to the projective transform may be enough of a save that we can add the iterative distortion inverse in #436 and be about the same speed as before. It's hard to say. I think you are right though that this is related to the above discussion
